examples/image.rs

```diff
@@ -99,7 +99,14 @@ async fn run(
     let path = &format!("{}:{}", &image, &version).replace("/", "_");
     let path = std::path::Path::new(&path);
     if path.exists() {
-        return Err(format!("path {:?} already exists, exiting", &path).into());
+        let msg = format!("path {:?} already exists", &path);
```

Comment:
What about moving all this path cleaning/creation outside of the `async` block into its own function? It is blocking logic anyway and it is a bit distracting when readying through the registry-fetching logic. I think it can be safely performed upfront.